File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.java
class Solution {
    public String solution(String my_string, int[][] queries) {
        String answer = my_string;
        for (int[] query : queries) {
            int s = query[0];
            int e = query[1];
            String reverse_string = "";
            for (int i = e; i >= s; i--) {
                reverse_string += answer.charAt(i);
            }
            answer = answer.substring(0, s) + reverse_string + answer.substring(e + 1);
        }
        return answer;
    }
}

File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.py
def solution(my_string, queries):
    for query in queries:
        s = int(query[0])
        e = int(query[1])
        reversed_string = my_string[s:e+1][::-1]
        my_string = my_string[:s] + reversed_string + my_string[e+1:]
    
    return my_string